pkg/segment/utils/number_test.go

```diff
@@ -0,0 +1,396 @@
+package utils
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Test_isInvalid tests if the number is invalid. It should return invalidType.
+func Test_isInvalid(t *testing.T) {
+	n := Number{}
+	n.SetInvalidType()
+	assert.Equal(t, n.ntype(), invalidType)
+}
+
+// Test_invalidTypeReturnsErrorForInt64AndFloat64 tests converting invalidType to Int64 and Float64.
+func Test_invalidTypeReturnsErrorForInt64AndFloat64(t *testing.T) {
+	n := Number{}
+	n.SetInvalidType()
+
+	_, err := n.Int64()
+	assert.NotEqual(t, nil, err)
+
+	_, err = n.Float64()
+	assert.NotEqual(t, nil, err)
+
+}
+
+// Test_isBackfill tests if the number is a backfill. It should return 0 for both Int64 and Float64.
+func Test_isBackfill(t *testing.T) {
+
+	n := Number{}
+	n.SetBackfillType()
+	assert.Equal(t, n.ntype(), backfillType)
+
+	intVal, err := n.Int64()
+	assert.Equal(t, 0, intVal)
+	assert.Equal(t, nil, err)
+
+	floatVal, err := n.Float64()
+	assert.Equal(t, 0, floatVal)
+	assert.Equal(t, nil, err)
+}
+
+// Test_isInt64 tests if the number is an int64. It should return set value for Int64 and an error for Float64.
+func Test_isInt64(t *testing.T) {
+
+	n := Number{}
+	n.SetInt64(42)
+	assert.Equal(t, int64Type, n.ntype())
+
+	val, err := n.Int64()
+	assert.Equal(t, 42, val)
+	assert.Equal(t, nil, err)
+
+	_, err = n.Float64()
+	assert.NotEqual(t, nil, err)
+
+}
+
+// Test_isFloat64 tests if the number is a float64. It should return set value for Float64 and an error for Int64.
+func Test_isFloat64(t *testing.T) {
+
+	n := Number{}
+	n.SetFloat64(42.0)
+	assert.Equal(t, float64Type, n.ntype())
+
+	val, err := n.Float64()
+	assert.Equal(t, 42.0, val)
+	assert.Equal(t, nil, err)
+
+	_, err = n.Int64()
+	assert.NotEqual(t, nil, err)
+
+}
+
+// Test_reset tests resetting the number. It should return invalidType.
+func Test_reset(t *testing.T) {
+
+	n := Number{}
+	n.SetInt64(42)
+	assert.Equal(t, int64Type, n.ntype())
+	n.Reset()
+	assert.Equal(t, invalidType, n.ntype())
+
+}
+
+// Test_constructor tests the constructor of the number. It should return the correct type.
+func Test_constructor(t *testing.T) {
+
+	n := Number{}
+	assert.Equal(t, invalidType, n.ntype())
+
+}
+
+// Test_numberSetAndGet tests setting and getting the number. It should return the correct value.
+func Test_numberSetAndGet(t *testing.T) {
+	n := Number{}
+
+	// Test Int64
+	n.SetInt64(42)
+	if val, err := n.Int64(); err != nil || val != 42 {
+		assert.Fail(t, "Expected 42, got %v with error %v", val, err)
+	}
+
+	// Test Float64
+	n.SetFloat64(3.14)
+	if val, err := n.Float64(); err != nil || math.Abs(val-3.14) > 1e-6 {
```

Comment:
Can you not directly do equal? Since `3.14` is also float?

Also use: `assert.Nil()`, `assert.Equal()`, `assert.NotNil()`

Reply:
Changed all places where condition was used to appropriate asserts

Reply:
using abs value because of the float precision error. It works without checking precision, but this makes sure it handles the possible issues as well.